app/api/templates/templates.ts

```diff
@@ -1,3 +1,4 @@
+/* eslint-disable max-statements */
```

Comment:
I think it's ok to move the `shouldUpdateMetadata` logic to an outside function, to avoid this.

Reply:
Removed